src/object_ref.jl

```diff
@@ -112,23 +108,17 @@ function _register_ownership(obj_ref::ObjectRef, outer_obj_ref::Union{ObjectRef,
         ray_jll.FromNil(ray_jll.ObjectID)
     end
 
-    # we've overloaded getproperty for this one to create the actual owner ref
-    owner_address = obj_ref.owner_address
-
     worker = ray_jll.GetCoreWorker()
-    if !isnothing(obj_ref.owner_address) && !has_owner(obj_ref)
+    if !has_owner(obj_ref)
```

Comment:
I know it's not the point of this PR, but while we're touching this code, remind me again why we need to check for `!has_owner` here?  IIRC the java code always registers ownership on deserialization...

Reply:
IIRC the reason the `has_owner` check was originally added was during the time we were having the nasty problem around registration. I believe the thinking was that possibly registering multiple times could cause issues in the backend so let's avoid doing it. I think we can safely remove this check but let's do that as a follow up